CMakeLists.txt

```diff
@@ -1571,7 +1571,7 @@ if(FLAKE8_FOUND AND FLAKE8_EXECUTABLE)
 endif()
 
 # ========================== java ==========================
-if(BUILD_JAVA OR BUILD_opencv_java)
```

Comment:
`BUILD_opencv_XXX` are widely used CMake variables (they are defined for all modules like `BUILD_opencv_highgui`).

`BUILD_JAVA=ON` doesn't not guarantee that all dependencies are found (JDK and Ant) and `BUILD_opencv_java=ON`

Reply:
So `BUILD_opencv_java` is more reliable option? I mean if `java` is OpenCV module it'd be better to keep convenient  `BUILD_opencv_java`.

Reply:
`BUILD_JAVA` behaves like `WITH_XXX` options (in case of turning off it avoids unnecessary JDK/Ant searching).
`BUILD_opencv_java` stays here (as OpenCV modules variable).